### PR TITLE
#400: Project Edit Form

### LINKED ISSUE #398: Transition to MUI ProjectEditContainer.tsx (Container, Form)
Transition ProjectEditContainer.tsx to material UI. Remove all react-bootstrap imports. Make visuals look as similar to before as possible. Fix any weird issues that switching to the different component causes.

### LINKED ISSUE #399: Transition to MUI ProjectEditDetails.tsx (Col, Container, Row, Form, InputGroup)
Transition ProjectEditDetails.tsx to material UI. Remove all react-bootstrap imports. Make visuals look as similar to before as possible. Fix any weird issues that switching to the different component causes.

### LINKED ISSUE #400: Transition to MUI ProjectEditSummary.tsx (Form)
Transition ProjectEditSummary.tsx to material UI. Remove all react-bootstrap imports. Make visuals look as similar to before as possible. Fix any weird issues that switching to the different component causes.

### PR DESCRIPTION
**Pull request by anthonybernardi**
## Changes

Changed the entire form page to use react hook forms which is a lot better and it fully works. I did absolutely no work to make it look nice though. This was already a ton of work so I might not want to do too much of that but idk

## Notes

Made a custom react hook text field component which is super useful. Also just ignored the EditableText component thing since so much of it needed to be replace for React Hook Forms. Unfortunately, that leads to some features going away like pressing enter makes a new text box and stuff like that (although we can work on readding that later)

## Screenshots

NOTE: see most recent comment for updated screenshots

Screen Shot 2022-11-09 at 6 12 51 PM
Screen Shot 2022-11-09 at 6 13 01 PM

## To Do

- finish the yup schema and make it show the errors somehow

Closes #398 
Closes #399 
Closes #400

### CODE REVIEW

File: src/frontend/src/components/ReactHookTextField.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  size?: 'small' | 'medium';
+  sx?: any;
+  type?: any;
+  icon?: any;
+  multiline?: boolean;
+  rows?: number;
+  errorMessage?: FieldError;
+}
+
+const ReactHookTextField: React.FC<ReactHookTextFieldProps> = ({
```
Comment (RChandler234) [inline src/frontend/src/components/ReactHookTextField.tsx:24]: Love this

**Review by RChandler234 — CHANGES_REQUESTED** [review]
Love all the small stuff you cleaned up, splitting out the ProjectEditContainer into separate files where possible is the main comment and also just making the project details section look better. Also if you could put some space between the submit and cancel buttons at the bottom, or do it overlayed like it was before. Looks a lot better than it did already!

This is a picture of the old project edit form for reference:
image

File: src/frontend/src/pages/ProjectDetailPage/ProjectEditContainer/ProjectEditContainer.tsx [hunk 2]
```diff
@@ -3,191 +3,172 @@
...
+      new URL(url);
+      return true;
+    } catch (_) {
+      return false;
+    }
+  }
+  return false;
+};
+
+const buildEditableBulletList = (
```
[the remark this answers is not in the document]
Comment (anthonybernardi) [reply src/frontend/src/pages/ProjectDetailPage/ProjectEditContainer/ProjectEditContainer.tsx:44]: tbh just wanted to see how short i could get it with react hook forms and also didn't really care

**Comment by anthonybernardi** [comment]
New pictures: 
Screen Shot 2022-11-13 at 4 50 53 PM
Screen Shot 2022-11-13 at 4 50 57 PM
Screen Shot 2022-11-13 at 4 51 05 PM
Screen Shot 2022-11-13 at 4 51 08 PM

File: src/frontend/src/layouts/PageBlock.tsx [hunk 3]
```diff
@@ -29,7 +29,7 @@ const PageBlock: React.FC<PageBlockProps> = ({ title, headerRight, children }) =
   return (
     <Card sx={{ my: 2, background: theme.palette.background.paper }} variant="outlined">
       <CardContent>
-        <Box sx={{ display: 'flex', flexDirection: 'row' }}>
+        <Box sx={{ display: 'flex', flexDirection: 'row', mb: 1 }}>
```
Comment (RChandler234) [inline src/frontend/src/layouts/PageBlock.tsx:32]: what does mb do again?

File: src/frontend/src/components/ReactHookEditableList.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  remove: UseFieldArrayRemove;
+}
+
+const ReactHookEditableList: React.FC<ReactHookEditableListProps> = ({ name, ls, register, append, remove }) => {
+  return (
+    <>
+      {ls.map((_element, i) => {
+        return (
+          <Grid item sx={{ display: 'flex', alignItems: 'center' }}>
+            <TextField required sx={{ width: 9 / 10 }} {...register(`${name}.${i}.detail`)} />
```
Comment (RChandler234) [inline src/frontend/src/components/ReactHookEditableList.tsx:19]: need to add autoComplete = "off" like you have in ReactHookTextField

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/frontend/src/layouts/PageBlock.tsx (hunk 3 above)
Comment (anthonybernardi) [reply src/frontend/src/layouts/PageBlock.tsx:32]: margin bottom

**Review by RChandler234 — APPROVED** [review]

**Pull request merged**